rockcraft/oci.py

```diff
@@ -156,15 +156,62 @@ def digest(self) -> bytes:
         parts = output.split(":", 1)
         return bytes.fromhex(parts[-1])
 
+    def set_entrypoint(self, entrypoint: List[str]) -> None:
+        """Set the OCI image entrypoint.
+
+        :param entrypoint: The default list of arguments to use as the command to
+            execute when the container starts.
+        """
+        emit.progress("Configuring entrypoint...")
+        image_path = self.path / self.image_name
+        params = ["--clear=config.entrypoint"]
+        for entry in entrypoint:
+            params.extend(["--config.entrypoint", entry])
+        _config_image(image_path, params)
+        emit.message(f"Entrypoint set to {entrypoint}", intermediate=True)
+
+    def set_cmd(self, cmd: List[str]) -> None:
+        """Set the OCI image default command parameters.
+
+        :param cmd: The default arguments to the entrypoint of the container. If
+            entrypoint is not defined, the first entry of cmd is the executable to
+            run when the container starts.
+        """
+        emit.progress("Configuring cmd...")
+        image_path = self.path / self.image_name
+        params = ["--clear=config.cmd"]
+        for entry in cmd:
+            params.extend(["--config.cmd", entry])
+        _config_image(image_path, params)
+        emit.message(f"Cmd set to {cmd}", intermediate=True)
+
+    def set_env(self, env: List[str]) -> None:
```

Comment:
could `env` be a `dict` instead? eg.:

```
env:
  FOO: bar
  other_var: 42
```

Reply:
We could express it as a dict with a few caveats: the YAML spec doesn't guarantee order (but our implementation will preserve order in Python 3.7+), and as a dictionary it won't allow constructs such as:
```
env:
   FOO: bar
   FOO: $FOO:baz
```
(unsure if docker allows this type of construct however).

This was addressed in Snapcraft by using a list of single-entry dictionaries, such as:
```
env:
  - FOO: bar
  - FOO: $FOO:baz
```
Using a similar construct in rockcraft would also make it consistent with `build-environment` in parts, which is a good thing IMHO.

Reply:
a list is fine too

Reply:
Changed to use the same format used in snapcraft (list of dictionaries).

Reply:
We probably want the duplicate keys detection we added to the YAML loader for Snapcraft for this too.